controllers/PollResponseController.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * @file Controller RESTful Web service API for pollResponses resource
+ */
+import { Request, Response, Express } from "express";
+import PollResponseDao from "../daos/PollResponseDao";
+import PollResponseControllerI from "../interfaces/PollResponseControllerI";
+import PollResponse from "../models/PollResponse";
+import PollController from "./PollController";
+
+/**
+ * @class PollResponseController Implements RESTful Web service API for pollResponses resource.
+ * Defines the following HTTP endpoints:
+ * <ul>
+ *     <li>POST /api/user/:uid/response/polls/:pid to create a new pollResponse instance for
+ *     a given user and poll</li>
+ *     <li>GET /api/users/:uid/pollsresponded to retrieve a particular pollResponse instances from a specified user</li>
+ *     <li>GET /api/usersrepsonded/polls/:pid to retrieve all users pollResponse for certain poll </li>
+ *     <li>GET /api/polls/:pid/ to retrieve the pollResponse instances for a given poll id</li>
+ *     <li>PUT /api/users/:uid/poll/:pid to modify an individual pollResponse instance </li>
+ *     <li>DELETE /api/users/:uid/deleteresponse/polls/:pid to remove a particular pollResponse instance from
+ *     specified user and poll</li>
+ * </ul>
+ * @property {PollResponseDao} pollResponseDao Singleton DAO implementing pollResponse CRUD operations
+ * @property {PollResponseController} pollResponseController Singleton controller implementing
+ * RESTful Web service API
+ */
+export default class PollResponseController implements PollResponseControllerI {
+    private static pollResponseDao:PollResponseDao = PollResponseDao.getInstance();
+    private static pollResponseController: PollResponseController | null = null;
+
+    private constructor() {}
+
+    /**
+     * Creates singleton controller instance
+     * @param {Express} app Express instance to declare the RESTful Web service API
+     * @return PollResponseController
+     */
+    public static getInstance = (app: Express): PollResponseController => {
+        if(PollResponseController.pollResponseController == null) {
+            PollResponseController.pollResponseController = new PollResponseController();
+
+            //Restful User Web service API
+            app.get('/api/users/:uid/pollsresponded',PollResponseController.pollResponseController.findPollResponsesByUser);
+            app.get('/api/usersrepsonded/polls/:pid',PollResponseController.pollResponseController.findAllUsersReplyPollResponse);
+            app.get('/api/polls/:pid',PollResponseController.pollResponseController.findPollResponseByPollId);
```

Comment:
Do we need this api? This was not part of initial API discussions. If we need this then we need to change the api path as it clashes with other TCreator API

Reply:
change it to different api